components/psst/browser/core/psst_rule.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/flat_set.h"
+#include "base/files/file_path.h"
+#include "base/json/json_value_converter.h"
+#include "base/values.h"
+#include "extensions/common/url_pattern_set.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+class GURL;
+
+namespace psst {
+
+struct MatchedRule {
+  std::string test_script;
+  std::string policy_script;
+  int version;
+};
+
+class PsstRule {
+ public:
+  PsstRule();
+  ~PsstRule();
+
+  // Registers the mapping between JSON field names and the members in this
+  // class.
+  static void RegisterJSONConverter(
+      base::JSONValueConverter<PsstRule>* converter);
+  static absl::optional<std::vector<std::unique_ptr<PsstRule>>> ParseRules(
+      const std::string& contents);
+  static bool GetURLPatternSetFromValue(const base::Value* value,
+                                        extensions::URLPatternSet* result);
+  static bool GetFilePathFromValue(const base::Value* value,
```

Comment:
why is this a static method instead of an anonymous method? Also why is it public? Same thing for `GetURLPatternSetFromValue`. Maybe they were static in debounce, etc... for tests? If so they should at least be private and add friend test